test/curated_cases.cc

```diff
@@ -0,0 +1,212 @@
+        const auto n_unified_sites = yaml["truth_unified_sites"];
+        S(parse_unified_sites(n_unified_sites, test_contigs));
+
+        // write truth gvcf
+        const auto n_truth_gvcf = yaml["truth_output_vcf"];
+        S(write_gvcfs(n_truth_gvcf, false))
+        return Status::OK();
+    }
+
+    // Verify that unified_site generated by GLnexus service matches
+    // the expected "truth"
+    Status check_unify_sites() {
+        Status s;
+        S(VCFData::Open(input_gvcfs, data));
+        s = Service::Start(*data, *data, svc);
+        REQUIRE(s.ok());
+
+        discovered_alleles als;
+        s = svc->discover_alleles("<ALL>", range(0, 0, 1000000), als);
+        REQUIRE(s.ok());
+
+        s =unified_sites(als, sites);
+        REQUIRE(s.ok());
+
+        sort(sites.begin(), sites.end());
+        sort(truth_sites.begin(), truth_sites.end());
+
+        if (sites != truth_sites) {
+            cout << "^^^^^ Unified sites differ from truth:";
+            cout << "Expected truth sites \n";
+            for (auto& site : truth_sites) {
+                cout << site.str() << endl;
+            }
+
+            cout << "Generated unified sites \n";
+            for (auto& site: sites) {
+                cout << site.str() << endl;
+            }
+        } else {
+            cout << "@@@@@ Unified sites agree!" << endl;
+        }
+        return Status::OK();
+    }
+
+    // Verify that the genotypes generated by GLnexus service matches
+    // the "truth" vcf output. Assumes that check_unified_sites has
+    // already been called, and sites have been populated.
+    Status check_genotypes() {
+        Status s;
+
+        // There should be at least 1 unified_site, this will fail
+        // if check_unified_sites() has not been called beforehand
+        REQUIRE(!sites.empty());
+
+        consolidated_loss losses;
+        string out_fn = "test/data/special_cases/" + name + "/output.vcf";
+        s = svc->genotype_sites(genotyper_config(), string("<ALL>"), sites, out_fn, losses, true);
+
+        REQUIRE(s.ok());
+
+        string diff_fn = "test/data/special_cases/" + name + "/output.diff";
+
+        string diff_cmd = "diff " + out_fn + " " + truth_vcf_fn + " > " + diff_fn;
+
+        int retval = system(diff_cmd.c_str());
+        REQUIRE (WIFEXITED(retval));
+
+        if (WEXITSTATUS(retval) == 0){
+            // no difference found in diff command
+            cout << "@@@@@ Output vcf and truth agrees!" << endl;
+        } else {
+            cout << "^^^^^ Output vcf and truth differs. Diff file: " << endl;
+            retval = system(("cat " + diff_fn).c_str());
+        }
+
+        cout << "++++++++++++++++++++++++++++++++++++" << endl;
+        cout << "End of test for Curated case: " << name << endl;
+        cout << "++++++++++++++++++++++++++++++++++++" << endl;
+        return Status::OK();
+    }
+
+    Status cleanup() {
+        // TO-DO: delete intermediate gvcf/vcf files written to disk
```

Comment:
Yea, I think we should write the temporary files under `/tmp`, and try to clean them up at the end. The easiest way to handle the cleanup is probably to write them all into a specially created subdirectory, and then shell out to `rm -rf` at the end.